packages/slice-machine/lib/models/common/tracker.ts

```diff
@@ -1,38 +1,93 @@
 import Analytics from "analytics-node";
 import { LibraryUI } from "./LibraryUI";
 
-export default class Tracker {
+export interface TrackerOptions {
+  flushAt?: number | undefined;
+  flushInterval?: number | undefined;
+  host?: string | undefined;
+  enable?: boolean | undefined;
+  timeout?: number | string | undefined;
+}
+
+// the default tracker should do nothing if parts are missing, it's a placeholder to prevent runtime errors in the meantime
+const DEFAULT_TRACKER: Tracker = {
+  libraries() {},
+  resolveUser: () => {},
+};
+
+export class TrackerBuilder {
   readonly analytics: Analytics;
   readonly slicemachineVersion: string;
 
   constructor(
     readonly key: string | undefined,
-    readonly smVersion: string | undefined
+    readonly smVersion: string | undefined,
+    readonly credentials: {
+      readonly userId?: string | undefined;
+      readonly anonymousId?: string | undefined;
+    },
+    readonly repo: string | undefined,
+    readonly options: TrackerOptions
   ) {
     if (!key) throw new Error("Missing write key for tracker.");
+    console.log({ credentials });
 
-    this.analytics = new Analytics(key);
+    this.analytics = new Analytics(key, this.options);
     this.slicemachineVersion = smVersion || "default"; //default will be logged if we can't parse package.json
   }
 
-  Repository = (repo: string | void) => {
-    if (!repo) return; // we can't track as long as we're not tied to a repo.
+  build: () => Tracker = () => {
```

Comment:
As we have a class why don't expose directly the functions `resolveUser` and `libraries` ? Instead of retruning a "fake tracker" w can throw error or early return on every functions no ?

Reply:
I still want a tracker anyway because at some point we'll track even if we're not repo related but I don't want to throw an error if something is missing. So in the meantime I did it this way but it's not ideal. And I don't want conditions on params on every single tracking function. It definitely needs some refinement